### ISSUE TITLE
large_smoke particle generates error on Paper/Spigot 1.20.6

### ISSUE THREAD

**Issue by SlimeDog** [SkriptLang/Skript#6635]
### Skript/Server Version

```markdown
>skript info
[16:00:09] [Server thread/INFO]: [Skript] Skript's aliases can be found here: https://github.com/SkriptLang/skript-aliases
[16:00:09] [Server thread/INFO]: [Skript] Skript's documentation can be found here: https://docs.skriptlang.org/
[16:00:09] [Server thread/INFO]: [Skript] Skript's tutorials can be found here: https://docs.skriptlang.org/tutorials
[16:00:09] [Server thread/INFO]: [Skript] Server Version: 4137-Spigot-e2c1eee-666f091 (MC: 1.20.6)
[16:00:09] [Server thread/INFO]: [Skript] Skript Version: 2.8.5 (skriptlang-github)
[16:00:09] [Server thread/INFO]: [Skript] Installed Skript Addons: None
[16:00:09] [Server thread/INFO]: [Skript] Installed dependencies: 
[16:00:09] [Server thread/INFO]: [Skript]  - Vault v1.7.3-b131
[16:00:09] [Server thread/INFO]: [Skript]  - WorldGuard v7.0.10-SNAPSHOT+2263-c65dd83
```

### Bug Description

The following constructs fail to parse on Paper/Spigot 1.20.5 or 1.20.6
```
player's balance
money of {_player}
set {_PARTICLE} to large smoke
```
**UPDATE: The economy constructs work correctly when Essentials is installed as the economy provider.**

### Expected Behavior

Those constructs used to parse correctly, prior to Paper/Spigot 1.20.5 

### Steps to Reproduce

- Update to Paper or Spigot 1.20.6
- Run Skript 2.8.4 or 2.8.5
- Start server
- Read logs

### Errors or Screenshots

```
[15:51:15] [Server thread/INFO]: [Skript] Loaded 10 variables in 0.0 seconds
[15:51:16] [Server thread/INFO]: [Skript] Line 17: (feed-me.sk)
[15:51:16] [Server thread/INFO]:     Can't understand this expression: 'the player's balance'
[15:51:16] [Server thread/INFO]:     Line: set {_balance} to the player's balance
[15:51:16] [Server thread/INFO]:
[15:51:16] [Server thread/INFO]: [Skript] Line 51: (feed-me.sk)
[15:51:16] [Server thread/INFO]:     Can't understand this condition/effect: remove {_cost} from the player's balance
[15:51:16] [Server thread/INFO]:     Line: remove {_cost} from the player's balance
[15:51:16] [Server thread/INFO]:
[15:51:16] [Server thread/INFO]: [Skript] Line 100: (grant-world-access.sk)
[15:51:16] [Server thread/INFO]:     Can't understand this expression: 'money of {_player}'
[15:51:16] [Server thread/INFO]:     Line: set {_BALANCE} to money of {_player}
[15:51:16] [Server thread/INFO]:
[15:51:16] [Server thread/INFO]: [Skript] Line 112: (grant-world-access.sk)
[15:51:16] [Server thread/INFO]:     Can't understand this condition/effect: subtract {_cost} from money of {_player}
[15:51:16] [Server thread/INFO]:     Line: subtract {_cost} from money of {_player}
[15:51:16] [Server thread/INFO]:
[15:51:17] [Server thread/INFO]: [Skript] Line 286: (test.sk)
[15:51:17] [Server thread/INFO]:     Can't understand this expression: 'large smoke'
[15:51:17] [Server thread/INFO]:     Line: set {_PARTICLE} to large smoke
[15:51:17] [Server thread/INFO]:
[15:51:17] [Server thread/INFO]: [Skript] Line 318: (test.sk)
[15:51:17] [Server thread/INFO]:     Can't understand this expression: 'money of {_player}'
[15:51:17] [Server thread/INFO]:     Line: set {_VALUE} to money of {_player}
[15:51:17] [Server thread/INFO]:
[15:51:17] [Server thread/INFO]: [Skript] Line 22: (on-join.sk)
[15:51:17] [Server thread/INFO]:     Can't understand this condition/effect: set player's balance to 1000
[15:51:17] [Server thread/INFO]:     Line: set player's balance to 1000
```

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this report.

**Comment by APickledWalrus** [comment]
Is vault/your economy provider working correctly on those versions? Some particles were removed so it may be that large smoke is one of them.

**Comment by SlimeDog** [comment]
According to my logs, `Vault` is working correctly.
```
grep -i Vault logs/latest.log
[15:50:24] [Server thread/INFO]: [Vault] Loading server plugin Vault v1.7.3-b131
[15:50:33] [Server thread/INFO]: [Vault] Enabling Vault v1.7.3-b131
[15:50:34] [Server thread/INFO]: [Vault] [Permission] SuperPermissions loaded as backup permission system.
[15:50:34] [Server thread/INFO]: [Vault] Enabled Version 1.7.3-b131
[15:50:34] [Server thread/INFO]: [LuckPerms] Registered Vault permission & chat hook.
[15:50:34] [Server thread/INFO]: WEPIF: Vault detected! Using Vault for permissions
[15:50:47] [Server thread/INFO]: [Citizens] Loaded economy handling via Vault.
[15:51:09] [Server thread/INFO]: [Disease] Vault plugin Found
[15:51:09] [Server thread/INFO]: [Disease] Vault Enabled
[15:51:09] [Server thread/INFO]: [Sentinel] Vault linked! Group targets will work.
[15:51:12] [Server thread/INFO]: [CommandSigns] Plugin vault detected
[15:51:12] [Server thread/INFO]: [CommandSigns] No vault economy hooked.
[15:51:18] [Server thread/INFO]: [PlaceholderAPI] Successfully registered external expansion: vault [1.8.3]
[15:51:29] [Server thread/INFO]: [PlaceholderAPI] Successfully registered external expansion: vault [1.8.3]
```

I find no evidence in various places that `large_smoke` has been removed.

**Comment by APickledWalrus** [comment]
Thanks for the logs. There were many internal changes with 1.20.5/1.20.6 so these may have been something missed on our end. I'll look into it this week and get back to you :)

**Comment by SlimeDog** [comment]
`/particle minecraft:large_smoke` is recognized on Spigot and Paper 1.20.6.

**Comment by APickledWalrus** [comment]
Yep I confirmed it's still present in the particle entries. Some particles were removed so it could be something going wrong in the particle system.

**Comment by ShaneBeee** [comment]
could you please provide the output of the command `/vault-info`
(this will detail which eco plugin you're using)

I tested on my end:
```
> vault-info
[22:36:18 INFO]: [Vault] Vault v1.7.3-b131 Information
[22:36:18 INFO]: [Vault] Economy: BeeConomy [BeeConomy]
[22:36:18 INFO]: [Vault] Permission: SuperPerms [SuperPerms]
[22:36:18 INFO]: [Vault] Chat: None [null]
```
<img width="503" alt="Screenshot 2024-05-01 at 10 36 55 PM" src="https://github.com/SkriptLang/Skript/assets/20008482/e453c324-4d18-47a9-a708-2d8493a05dd3">

I cannot replicate said issue.

**Comment by SlimeDog** [comment]
```
>vault-info
[22:48:33] [Server thread/INFO]: [Vault] Vault v1.7.3-b131 Information
[22:48:33] [Server thread/INFO]: [Vault] Economy: None [null]
[22:48:33] [Server thread/INFO]: [Vault] Permission: LuckPerms [LuckPerms, SuperPerms]
[22:48:33] [Server thread/INFO]: [Vault] Chat: LuckPerms [LuckPerms]
```
My economy plugins is Essentials, but Essentials is still broken on MC 1.20.5. When it is stable, that will probably resolve the `balance` issues.

**Comment by Fusezion** [comment]
Don't think I read it right on discord but 
```
[22:48:33] [Server thread/INFO]: [Vault] Economy: None [null]
```
that syntax won't work if skript has nothing to work with

**Comment by SlimeDog** [comment]
> Don't think I read it right on discord but
> 
> ```
> [22:48:33] [Server thread/INFO]: [Vault] Economy: None [null]
> ```
> 
> that syntax won't work if skript has nothing to work with

As noted in my previous comment ... mea culpa. When I loaded Essentials (although broken), it worked enough to satisfy Vault and the Skript economy commands.

**Comment by SlimeDog** [comment]
Error is still present in Skript 2.9.0-beta1

**Comment by sovdeeth** [comment]
> Error is still present in Skript 2.9.0-beta1

yeah the beta doesn't have any additional bug fixes compared to 2.8.5
at least none related to 1.20.5+

**Comment by APickledWalrus** [comment]
I've completed my investigation and begun work on a fix. There were just a lot of changes to the Particle Enum that were not addressed in the initial 1.20.5 support PR. I'll have a PR open soon.

**Comment by APickledWalrus** [comment]
I've opened PR https://github.com/SkriptLang/Skript/pull/6716 which should resolve this issue. It is based on 2.8.5 (really, the upcoming 2.8.6 which just includes a few other fixes)

If you test it, let me know if you have any issues. You can download the artifact from here: https://github.com/SkriptLang/Skript/actions/runs/9212689747 (direct link: https://github.com/SkriptLang/Skript/actions/runs/9212689747/artifacts/1532145115)

**Comment by SlimeDog** [comment]
I'm running Skript 2.9.0-beta1. I am not included to go back, unless you advise.

**Comment by APickledWalrus** [comment]
I might be able to make a version that is based on 2.9. I'll update here when I am able to.

**Comment by APickledWalrus** [comment]
This should be fixed in 2.8.6. It will be included in the dev/feature (2.9.0) builds soon.

**Comment by SlimeDog** [comment]
Decided to revert. Confirmed fixed in Skript 2.8.7. Thanks.

**Comment by APickledWalrus** [comment]
Happy to hear it's working. I've gone ahead and updated the `dev/feature` branch (upcoming 2.9) to be based on 2.8.7, so this fix will be present there if you choose to download the nightly build.
